android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboard.java

```diff
@@ -653,6 +630,29 @@ public boolean setKeyboard(String packageID, String keyboardID, String languageI
     return retVal;
   }
 
+  public void showBanner(boolean flag) {
+    String jsString = KMString.format("showBanner(%b)", flag);
+    loadJavascript(jsString);
+  }
+
+  public KMManager.BannerType getBanner() {
+    return currentBanner;
+  }
+
+  public void setBanner(KMManager.BannerType bannerType) {
+    currentBanner = bannerType;
+  }
+
+  public String getHTMLBanner() {
+    return this.htmlBannerString;
+  }
+
+  public void setHTMLBanner(String contents) {
+    this.htmlBannerString = contents;
+    String jsString = KMString.format("setBannerHTML('%s')", contents);
```

Comment:
Will this work with arbitrary HTML, even HTML that contains `'` characters? i.e. are they escaped?

Reply:
Handled now with JSONObject.quote()